File: lesson5/python3/homework/homework5_2.py
#!/usr/bin/python3
#: lesson5/python3/homework/homework5_2.py


def divisors(numb_n):
    numb_r = numb_n
    for numb_d in range(numb_n, 0, -1):
        if numb_r % numb_d == 0:
            print("divisors ->", numb_d, end=' ')


numb_i = int(input("Введите число -> "))
divisors(numb_i)
print()

File: lesson5/c_c++/homework/homework5_2.c
//: lesson5/c_c++/homework/homework5_2.c
// Напишите процедуру, которая выводит на экран все делители переданного ей
// числа (в одну строку).
#include <stdio.h>

void divisors(int numb_i) {
    int numb_res = numb_i;
    while (numb_i > 0) {
        if (numb_res % numb_i == 0)
            printf("divisors -> %d ", numb_i);
        numb_i--;
    }
}

int main() {
    int numb_n;
    printf("Введите натуральное число -> ");
    scanf("%d", &numb_n);
    divisors(numb_n);
    printf("\n");
    return 0;
}
